clippy_lints/src/booleans.rs

```diff
@@ -416,8 +416,8 @@ fn simplify_not(cx: &LateContext<'_>, curr_msrv: Msrv, expr: &Expr<'_>) -> Optio
                 _ => None,
             }
             .and_then(|op| {
-                let lhs_snippet = lhs.span.get_source_text(cx)?;
-                let rhs_snippet = rhs.span.get_source_text(cx)?;
+                let lhs_snippet = lhs.span.source_callsite().get_source_text(cx)?;
+                let rhs_snippet = rhs.span.source_callsite().get_source_text(cx)?;
```

Comment:
You'll need `walk_span_to_context`, not `source_callsite`. You should be able to use `SyntaxContext::root()` as the target context.